rebasehelper/helpers/download_helper.py

```diff
@@ -0,0 +1,182 @@
+
+        def format_size(s):
+            units = [' ', 'K', 'M', 'G', 'T']
+            i = 0
+            while s >= 1024.0 and i < len(units) - 1:
+                s /= 1024.0
+                i += 1
+            return '{:>7.2F}{}'.format(s, units[i])
+
+        if download_total < 0:
+            # infinite progress bar
+            pct = ' ' * 4
+            pos = int(downloaded / infinite_step) % (bar_width - 5)
+            bar = '[{}<=>{}]'.format(' ' * pos, ' ' * (bar_width - 5 - pos))
+            ts = ' in {}'.format(format_time(delta))
+        else:
+            r = float(downloaded) / float(download_total) if download_total else 0.0
+            pct = '{:>3d}%'.format(int(r * 100))
+            pos = int(r * (bar_width - 3))
+            bar = '[{}>{}]'.format('=' * pos, ' ' * (bar_width - 3 - pos))
+            ts = 'eta {}'.format(format_time(delta / r - delta) if r > 0.0 else ' ' * 7 + '?')
+
+        size = format_size(downloaded)
+
+        # no point to log progress, write directly to stdout
+        sys.stdout.write('\r{}{}  {}  {} '.format(pct, bar, size, ts))
+        sys.stdout.flush()
+
+    @staticmethod
+    def request(url, **kwargs):
+        """Gets a Reponse object from HTTP, HTTPS or FTP URL.
+
+        Args:
+            url (str): URL from which the Response is obtained.
+            **kwargs: Optional arguments used to get Response object.
+
+        Returns:
+            requests.Response: Response object.
+
+        """
+
+        class FTPAdapter(requests.adapters.BaseAdapter):
+
+            def send(self, request, stream=False, timeout=None, verify=True,  # pylint: disable=unused-argument
+                     cert=None, proxies=None):  # pylint: disable=unused-argument
+                response = requests.models.Response()
+                response.request = request
+                response.connection = self
+                try:
+                    resp = urllib.request.urlopen(request.url)
+                except urllib.error.URLError as e:
+                    response.status_code = 400
+                    response.reason = e.reason
+                else:
+                    response.status_code = 200
+                    response.headers = requests.structures.CaseInsensitiveDict(getattr(resp, 'headers', {}))
+                    response.raw = resp
+                    response.url = resp.url
+                return response
+
+            def close(self):
+                pass
+
+        session = requests.Session()
+        session.mount('ftp://', FTPAdapter())
+
+        try:
+            return session.get(url, **kwargs)
+        except requests.exceptions.RequestException as e:
+            logger.error('%s: %s', type(e).__name__, six.text_type(e))
+            return None
+
+    @staticmethod
+    def download_file(url, destination_path, blocksize=8192):
+        """Downloads a file from HTTP, HTTPS or FTP URL.
+
+        Args:
+            url (str): URL from which the file is downloaded.
```

Comment:
This should be either `URL from which the file is to be downloaded.` or rather simple `URL to download.` or `URL to be downloaded.`.